### PR TITLE
Remove error from SDK AppGossip handler

### PR DESCRIPTION
**Pull request by joshua-kim**
## Why this should be merged

The `error` result in the SDK handler is unused. I think my original reasoning for this was that calling the handler might error so we should log the result but the concrete implementation can just opt to log this themselves.

The `error` needs to stay on the `*Request` handlers so that in the future we can send back an `AppRequestFailed` message once it's supported over p2p. 

## How this works

Removes the return value from `AppGossip`

## How this was tested

UTs

### CODE REVIEW

File: network/p2p/handler.go [hunk 1]
```diff
@@ -111,19 +109,9 @@ func (r *responder) AppRequest(ctx context.Context, nodeID ids.NodeID, requestID
...
-			zap.Uint64("handlerID", r.handlerID),
-			zap.Binary("message", msg),
-		)
-	}
-}
-
 func (r *responder) CrossChainAppRequest(ctx context.Context, chainID ids.ID, requestID uint32, deadline time.Time, request []byte) error {
-	appResponse, err := r.handler.CrossChainAppRequest(ctx, chainID, deadline, request)
+	appResponse, err := r.Handler.CrossChainAppRequest(ctx, chainID, deadline,
+		request)
```
Comment (dhrubabasu) [inline network/p2p/handler.go:114]: nit: can we revert this

**Comment by joshua-kim** [comment]
related: https://github.com/ava-labs/avalanchego/pull/2238

File: network/p2p/throttler_handler.go [hunk 2]
```diff
@@ -22,12 +22,12 @@ type ThrottlerHandler struct {
 	Throttler Throttler
 }
 
-func (t ThrottlerHandler) AppGossip(ctx context.Context, nodeID ids.NodeID, gossipBytes []byte) error {
+func (t ThrottlerHandler) AppGossip(ctx context.Context, nodeID ids.NodeID, gossipBytes []byte) {
 	if !t.Throttler.Handle(nodeID) {
-		return fmt.Errorf("dropping message from %s: %w", nodeID, ErrThrottled)
+		return
```
Comment (dhrubabasu) [inline network/p2p/throttler_handler.go:27]: Should we have logs here?

Comment (joshua-kim) [reply network/p2p/throttler_handler.go:32]: Yeah we can add a log here.

**Review by dhrubabasu — APPROVED** [review]

**Review by StephenButtolph — COMMENTED** [review]
How should we handle the coreth update? I believe this introduces a panic because we don't set `Log` in coreth yet right?

**Comment by joshua-kim** [comment]
> How should we handle the coreth update? I believe this introduces a panic because we don't set Log in coreth yet right?

Yeah I need to make a PR to coreth to include the log there, the log was only added in a recent revision which is why I haven't written a PR yet. We should probably merge in coreth first then the avalanchego PR.

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**